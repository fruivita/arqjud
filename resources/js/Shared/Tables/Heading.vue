<!--
    Heading comum (th) da tabela padrÃ£o.

    @link https://vuejs.org/guide/introduction.html
    @link https://tailwindcss.com/docs
    @link https://inertiajs.com/
 -->
<script setup>
const props = defineProps({
    fixo: { type: Boolean, default: false },
    texto: { type: String, default: '' },
});
</script>

<template>
    <th
        :class="{
            'sticky left-0 z-10 h-fit bg-primaria-200 dark:bg-secundaria-700': fixo,
        }"
        class="p-3"
    >
        <span v-if="texto" class="font-bold">{{ texto }}</span>

        <slot></slot>
    </th>
</template>
