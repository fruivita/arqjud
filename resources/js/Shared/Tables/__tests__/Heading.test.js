/**
 * Testes para o componente Heading.
 *
 * @link https://vuejs.org/guide/scaling-up/testing.html
 * @link https://test-utils.vuejs.org/guide/
 * @link https://vitest.dev/
 */

import Heading from '@/Shared/Tables/Heading.vue';
import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, test } from 'vitest';

let mountFunction;

beforeEach(() => {
    mountFunction = (options = {}) => {
        return mount(Heading, { ...options });
    };
});

describe('Heading', () => {
    // Caminho feliz
    test('propriedades do componente estão definidas', () => {
        expect(Heading.props).toMatchObject({
            fixo: { type: Boolean, default: false },
            texto: { type: String, default: '' },
        });
    });

    test('renderiza o componente respeitando o snapshot', () => {
        expect(
            mountFunction({
                props: { texto: 'foo' },
            }).html()
        ).toMatchSnapshot();
    });

    test('renderiza de maneira diversa o componente se ele for fixo respeitando o snapshot', () => {
        expect(
            mountFunction({
                props: { fixo: true },
            }).html()
        ).toMatchSnapshot();
    });

    test('não renderiza o texto do componente respeitando o snapshot', () => {
        expect(mountFunction({ slots: { default: 'foo' } }).html()).toMatchSnapshot();
    });
});
