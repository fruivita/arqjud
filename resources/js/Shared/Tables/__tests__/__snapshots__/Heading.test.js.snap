// Vitest Snapshot v1

exports[`Heading > n√£o renderiza o texto do componente respeitando o snapshot 1`] = `
"<th class=\\"p-3\\">
  <!--v-if-->foo
</th>"
`;

exports[`Heading > renderiza de maneira diversa o componente se ele for fixo respeitando o snapshot 1`] = `
"<th class=\\"sticky left-0 z-10 h-fit bg-primaria-200 dark:bg-secundaria-700 p-3\\">
  <!--v-if-->
</th>"
`;

exports[`Heading > renderiza o componente respeitando o snapshot 1`] = `"<th class=\\"p-3\\"><span class=\\"font-bold\\">foo</span></th>"`;
